### PR TITLE
Jcarter/performance improvement

### PR DESCRIPTION
**Pull request by jacobscarter**

### CODE REVIEW

**Comment by jacobscarter** [comment]
@StefanCodes Here is you PR, you will have to remove dist code, sorry about that.

**Comment by jacobscarter** [comment]
I'll have to look into that later tonight

**Comment by jacobscarter** [comment]
Also please note that this PR **removes** horizontal tree code, I didn't see any documentation on it and assumed support was dropped for it.

**Comment by jacobscarter** [comment]
@StefanCodes @Voles I would like to get involved in this project, it really is the best drag and drop tree option out there for angular.

I will start looking over Issues and see what I can contribute, I just want to make sure these PRs will be reviewed, commented on, and eventually merged. :)

Thanks.

**Comment by StefanCodes** [comment]
Thanks! I'll take a fresh look at this soon - there are a couple other PRs simmering that need to be merged as well.
Thanks for the offer to get involved - as you can see, we're short on contributors and every little scrap of help is greatly appreciated.  If you're interested in being added as a contributor, reach out to @Voles - I don't have the permissions necessary, but I can merge and release any PRs that come my way.

**Comment by jacobscarter** [comment]
@Voles if you're okay with it I am more than happy to help out, go ahead and give me contributor permissions if you'd like :)

**Comment by Voles** [comment]
@jacobscarter I've added you as a contributor. Thanks for your help!

It's best to remove the files in 'dist/' from PR's to prevent merge conflicts for other PR's.

**Comment by jacobscarter** [comment]
@StefanCodes @Voles added me as a contributor but I don't know how you guys feel about me merging my own PRs, I think as a best practice it would be better if one of you two merged in my PRs.  What do you think?

**Comment by StefanCodes** [comment]
@jacobscarter Welcome! I agree having someone else merge (or at least review) code is a good idea.

I'll have a look at this PR (hopefully tonight or tomorrow) and I'll document the review/merge/deploy process so we can all follow it consistently. There are enough little things to check that I think we'd be well  served by a checklist.

**Comment by StefanCodes** [comment]
I've reviewed most of the code and ran out of steam - I will finish this merge tomorrow!

**Comment by StefanCodes** [comment]
Merged and released in 2.18.0, thank you!

**Comment by StefanCodes** [comment]
Go figure. 2 minutes after pushing 2.18.0 I realized there was a regression and nuked the whole release.

As a result of this PR (specifically the "horizontal tree code"), I lost the ability to move nodes up and down the hierarchy (I guess that's horizontal movement). That's explicitly different from the feature you saw alluded to other places in the code that had to be with horizontal trees.

Unsettlingly, the tests didn't catch this and I missed it in a manual review. 

@jacobscarter do you want to take another look at the PR and see what got removed to break the drag-drop? You can reproduce it specifically by selecting the last node on the demo page and trying to move it up a level (to become a sibling of its parent), or by moving it down a level (to become a child of its sibling). I noted after reverting to 2.17 that it was working again.

We still need to release a 2.18 (there's at least one more PR that should go in), but my own tools are acting up and I've run out of patience today to fight with them.

**Comment by jacobscarter** [comment]
@StefanCodes you got it, I will review it tomorrow and get to the bottom of things.  On a side note in my experience the more simple you keep PRs and Releases the easier it is to avoid problems.  So I wouldn't worry about combining multiple PRs into one new release.  I don't think there is a problem with making a new minor version release with each PR, even if that means multiple increments in a day.  e.g. 2.18.0 with PR 1 and 10 minutes later when you merge PR 2 it is 2.19.0

Let me know what you think.

File: source/directives/uiTreeNode.js [hunk 1]
```diff
@@ -354,180 +411,175 @@
                     treeScope.resetEmptyElement();
                     treeScope = null;
                   }
+
+                  //Reset parent to source parent.
+                  dragInfo.resetParent();
                 }
 
-                // move horizontal
-                if (pos.dirAx && pos.distAxX >= config.levelThreshold) {
```
Comment (StefanCodes) [inline source/directives/uiTreeNode.js]: @jacobscarter I think this block (and a couple under it) are related to the regression. The horizontal movement referred to is related to horizontal mouse movement to change the nesting of a tree item, not the obsolete 'horizontal tree' concept.

Comment (jacobscarter) [inline source/directives/uiTreeNode.js]: Roger that, I will make an update to this and re-open PR shortly.

**Pull request merged**